tool/src/org/antlr/v4/codegen/model/OutputFile.java

```diff
@@ -32,10 +33,15 @@ public OutputFile(OutputModelFactory factory, String fileName) {
     }
 
 	public Map<String, Action> buildNamedActions(Grammar g) {
+		return buildNamedActions(g, null);
+	}
+
+	public Map<String, Action> buildNamedActions(Grammar g, Predicate<ActionAST> filter) {
 		Map<String, Action> namedActions = new HashMap<String, Action>();
 		for (String name : g.namedActions.keySet()) {
 			ActionAST ast = g.namedActions.get(name);
-			namedActions.put(name, new Action(factory, ast));
+			if(filter==null || filter.test(ast))
```

Comment:
I think the project style uses curly braces even if there's only one statement. It reduces one possibility of coding bugs.